libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/RouteUtils.java

```diff
@@ -178,22 +180,20 @@ public boolean isValidRouteProfile(String routeProfile) {
   @Nullable
   public BannerInstructions findCurrentBannerInstructions(LegStep currentStep, double stepDistanceRemaining) {
```

Comment:
What about extracting blocks of code into well-named (I see three clear parts here `Collections.sort`, the `for` loop and the `return` statement) `private` methods? We should keep `public` methods as clean as possible so they read like pseudocode (i.e. no conditionals, `for`s, etc.).